### PR TITLE
feat: /avm/res/network/virtual-wan

### LINKED ISSUE #550: [Module Proposal]: `avm/res/network/virtual-wan`
### Check for previous/existing GitHub issues/module proposals

- [X] I have checked for previous/existing GitHub issues/module proposals.

### Check this module doesn't already exist in the module indexes

- [X] I have checked for that this module doesn't already exist in the module indexes; or I'm proposing the module to be migrated from CARML/TFVM.

### Bicep or Terraform?

Bicep

### Module Classification?

Resource Module

### Module Name

avm/res/network/virtual-hub

### Module Details

Migrating virtual-wan from CARML to AVM

### Do you want to be the owner of this module?

Yes

### Module Owner's GitHub Username (handle)

arnoldna

### (Optional) Secondary Module Owner's GitHub Username (handle)

_No response_

### PR DESCRIPTION
**Pull request by arnoldna**
## Description

Migrate virtual-wan module from CARML to BRM.

Closes https://github.com/Azure/Azure-Verified-Modules/issues/550

[avm.res.network.virtual-wan](https://github.com/arnoldna/bicep-registry-modules/actions/workflows/avm.res.network.virtual-wan.yml)

## Adding a new module

- [ ] A proposal has been submitted and approved.
- [ ] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [ ] I have run `brm validate` locally to verify the module files.
- [ ] I have run deployment tests locally to ensure the module is deployable.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -106,10 +106,10 @@
 /avm/res/network/route-table/ @Azure/avm-res-network-routetable-module-owners-bicep @Azure/avm-core-team-technical-bicep
 #/avm/res/network/service-endpoint-policy/ @Azure/avm-res-network-serviceendpointpolicy-module-owners-bicep @Azure/avm-core-team-technical-bicep
 /avm/res/network/trafficmanagerprofile/ @Azure/avm-res-network-trafficmanagerprofile-module-owners-bicep @Azure/avm-core-team-technical-bicep
-#/avm/res/network/virtual-hub/ @Azure/avm-res-network-virtualhub-module-owners-bicep @Azure/avm-core-team-technical-bicep
+/avm/res/network/virtual-hub/ @Azure/avm-res-network-virtualhub-module-owners-bicep @Azure/avm-core-team-technical-bicep
```
Comment (eriqua) [inline .github/CODEOWNERS:109]: Please open 2 different PRs for the 2 modules

Comment (arnoldna) [reply .github/CODEOWNERS]: I added back the comment to virtual-hub since it is enabled in the other PR.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -109,7 +109,7 @@
 #/avm/res/network/virtual-hub/ @Azure/avm-res-network-virtualhub-module-owners-bicep @Azure/avm-core-team-technical-bicep
 /avm/res/network/virtual-network/ @Azure/avm-res-network-virtualnetwork-module-owners-bicep @Azure/avm-core-team-technical-bicep
 /avm/res/network/virtual-network-gateway/ @Azure/avm-res-network-virtualnetworkgateway-module-owners-bicep @Azure/avm-core-team-technical-bicep
-#/avm/res/network/virtual-wan/ @Azure/avm-res-network-virtualwan-module-owners-bicep @Azure/avm-core-team-technical-bicep
+/avm/res/network/virtual-wan/ @Azure/avm-res-network-virtualwan-module-owners-bicep @Azure/avm-core-team-technical-bicep
```
Comment (AlexanderSehr) [inline .github/CODEOWNERS:112]: Just blocking until it doesn't show errors anymore (presumably, the team is not yet ready)

File: .github/workflows/avm.res.network.virtual-hub.yml [hunk 3]
```diff
@@ -0,0 +1,83 @@
+name: "avm.res.network.virtual-hub"
```
Comment (AlexanderSehr) [inline .github/workflows/avm.res.network.virtual-hub.yml:1]: Seems to be an unrelated workflow from a different PR. I'd recommend to remove it from this branch

File: avm/res/network/virtual-wan/tests/e2e/waf-aligned/main.test.bicep [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  params: {
+    location: resourceLocation
+    name: '${namePrefix}${serviceShort}001'
+    allowBranchToBranchTraffic: true
+    allowVnetToVnetTraffic: true
+    disableVpnEncryption: true
+    lock: {
+      kind: 'CanNotDelete'
+      name: 'myCustomLockName'
+    }
```
Comment (AlexanderSehr) [inline avm/res/network/virtual-wan/tests/e2e/waf-aligned/main.test.bicep:60]:
```suggestion
```

I think it was utimately descided to remove the locks from the WAF-aligned tests. Thanks :)

File: avm/res/network/virtual-wan/version.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{
+    "$schema": "https://aka.ms/bicep-registry-module-version-file-schema#",
+    "version": "0.2",
```
Comment (AlexanderSehr) [inline avm/res/network/virtual-wan/version.json:3]:
```suggestion
    "version": "0.1",
```

File: .github/CODEOWNERS (hunk 1 above)
Comment (AlexanderSehr) [reply .github/CODEOWNERS]: Resolving as removed

**Review by AlexanderSehr — APPROVED** [review]
